src/stream/src/executor/managed_state/join/mod.rs

```diff
@@ -85,24 +88,15 @@ impl JoinRow {
             .collect_vec())
     }
 
-    /// Make degree as the last datum of row
-    pub fn into_row(mut self) -> Row {
-        self.row.0.push(Some(ScalarImpl::Int64(self.degree as i64)));
-        self.row
-    }
-
-    /// Convert [`Row`] with last datum as degree to [`JoinRow`]
-    pub fn from_row(row: Row) -> Self {
-        let mut datums = row.0;
-        let degree_datum = datums
-            .pop()
-            .expect("missing degree in JoinRow")
-            .expect("degree should not be null");
-        let degree = degree_datum.into_int64() as u64;
-        JoinRow {
-            row: Row(datums),
-            degree,
-        }
+    /// Return row and degree in `Row` format. The degree part will be inserted in degree table
+    /// later, so a pk prefix will be added.
+    ///
+    /// * `state_order_key_indices` - the order key of `row`
+    pub fn into_table_rows(self, state_order_key_indices: &[usize]) -> (Row, Row) {
+        let degree_datum = Some(ScalarImpl::Int64(self.degree as i64));
+        let pk_prefix = self.row.by_indices(state_order_key_indices);
+        let degree = state_table::append_pk_prefix(Row::new(vec![degree_datum]), pk_prefix);
+        (self.row, degree)
```

Comment:
I'm creating a schemaed degree by taking the order key of the row as prefix. We may optimize this.